test/shadowed-ripple.html

```diff
@@ -0,0 +1,270 @@
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Ripple in Shadow conditions</title>
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../../iron-test-helpers/mock-interactions.js"></script>
+
+  <link rel="import" href="../../polymer/polymer.html">
+  <link rel="import" href="../../iron-behaviors/iron-button-state.html">
+  <link rel="import" href="../paper-ripple-behavior.html">
+</head>
+<body>
+  <style>
+    #source {
+      height: 25px;
+      width: 50px;
+      background: green;
+    }
+  </style>
+
+  <dom-module id="sd-ripple">
+    <template>
+      <style>
+        :host {
+          display: block;
+          width: 200px;
+        }
+        #separate, #wrapper {
+          height: 50px;
+        }
+        #separate {
+          background: blue;
+        }
+        #wrapper {
+          background: red;
+        }
+      </style>
+      <div id="separate">
+        <div id="target">
+          Internal Text Node
+        </div>
+      </div>
+      <div id="wrapper">
+        <content id="content"></content>
+      </div>
+    </template>
+    <script>
+      HTMLImports.whenReady(function() {
+        Polymer({
+          is: 'sd-ripple',
+          behaviors: [
+            Polymer.IronButtonState,
+            Polymer.IronControlState,
+            Polymer.PaperRippleBehavior
+          ]
+        });
+      });
+  </script>
+  </dom-module>
+
+  <test-fixture id="basic">
+    <template>
+      <sd-ripple></sd-ripple>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="text">
+    <template>
+      <sd-ripple>Howdy!</sd-ripple>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="element">
+    <template>
+      <sd-ripple>
+        <div id="source">source!</div>
+      </sd-ripple>
+    </template>
+  </test-fixture>
+
+  <script>
+    suite('Paper Ripple Behavior - correct targeting', function() {
+      var ripple;
+
+      function check(host, node, expected, msg) {
+        var ripple = host.getRipple();
+        Polymer.dom.flush();
+        MockInteractions.down(node);
+        assert.equal(ripple.ripples.length > 0, expected, msg);
+        MockInteractions.up(node);
+      }
+
+      function checkLocation(host, node, location, expected, msg) {
+        var ripple = host.getRipple();
+        MockInteractions.down(node, location);
+        assert.equal(ripple.ripples.length > 0, expected, msg);
+        MockInteractions.up(node);
+      }
+
+      var timeout = 500;
```

Comment:
Yeah, I'm going to try and see if making each one a seperate test, rebuilding the fixture each time, is faster.